client-spec.md

```diff
@@ -0,0 +1,275 @@
+
+This approach has several advantages. First, signers no longer need to manage signing keys; they can generate them fresh for each signature. Second, the risk of a leaked signing key is lower: after the validity period expires, the key cannot be used to sign any payloads without the cooperation of the timestamping service. Finally, artifact lifetime and expiration can be managed independently of key lifetime.
+
+In this approach, the certificate authority and timestamping services are trusted parties. To mitigate the security risks of centralization, we can introduce accountability in the form of *transparency*: public logs of all activity (certificates and signatures) that can be monitored for misbehavior. We implement this transparency property with a Certificate Transparency (CT) log ([RFC 6962](https://datatracker.ietf.org/doc/html/rfc6962); see [Spec: Fulcio](https://github.com/sigstore/architecture-docs/blob/main/fulcio-spec.md) for details on its integration with the identity service) and a transparency service ([Spec: Transparency Service](https://docs.google.com/document/u/0/d/1NQUBSL9R64\_vPxUEgVKGb0p81\_7BVZ7PQuI078WFn-g/edit)). The certificate authority will submit certificates to a CT log, and the signing client will submit payload metadata to the transparency service.
+
+This document describes this flow in detail.
+
+### 1.1. Requirements Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14 \[[RFC2119](https://www.rfc-editor.org/info/rfc2119)\] \[[RFC8174](https://www.rfc-editor.org/info/rfc8174)\] when, and only when, they appear in all capitals, as shown here.
+
+### 1.2. Terminology
+
+**Payload.** The payload is the data to be signed (represented as a sequence of bytes). This might be a digital artifact, or an attestation/claim/metadata about a digital artifact.
+
+**Identity.** A compact, virtual representation of an external actor, like a human or computer—for instance, a person's account on a web service, or the name of a computer system in a production environment.
+
+**Verification material.** All data required to verify a signature on a payload, including the signature itself and materials from the various services.
+
+**Client.** The software implementation used to sign and verify in Sigstore.
+
+### 1.3. Parties
+
+**Signer.** An entity who wishes to sign a payload using an identity they control.
```

Comment:
"signer" is used in various sections, such as §2, §2.1.2, §2.1.5, where "client" might be more appropriate based on the terminology and party definitions. It might be helpful if we review these uses or consider redefining the terms for consistency throughout the doc

Reply:
Opened #17 to track and address after we merge this.